homeassistant/components/homematicip_cloud/binary_sensor.py

```diff
@@ -136,6 +136,44 @@ async def async_setup_entry(
         async_add_entities(entities)
 
 
+class HomematicipCloudConnectionSensor(HomematicipGenericEntity, BinarySensorEntity):
+    """Representation of the HomematicIP cloud connection sensor."""
+
+    def __init__(self, hap: HomematicipHAP) -> None:
+        """Initialize the cloud connection sensor."""
+        super().__init__(hap, hap.home, "Cloud Connection")
+
+    @property
+    def device_info(self) -> Dict[str, Any]:
+        """Return device specific attributes."""
+        # Adds a sensor to the existing HAP device
+        return {
+            "identifiers": {
+                # Serial numbers of Homematic IP device
```

Comment:
you dont pass device to entity constructor so this seems like _device would error

Reply:
No that that ok. For this special entity the hmip home is the device and saved in _device.

Reply:
I switched the implementation from _device to _home.
Both contain in this case the same object, but it looks better.

Reply:
ok, might need to do similiar for unique_id here because `return f"{self.__class__.__name__}_{self._device.id}"` in generic entity is going to access _device

Reply:
As mentioned above this is OK, because hmip home is the device and saved in _device. There is no need to have an extra unique_id.